### PR TITLE
[RFC] Skip test execution when during IFS wait time

### PR DESCRIPTION
**Pull request by AthenasJimenez**

### CODE REVIEW

**Comment by AthenasJimenez** [comment]
Something that I'm still unable to do is marking test result as "skip" instead of "pass". Any way that I could do this?

```
- test: ifs
  time-at-start: { elapsed:      0.000, now: !!timestamp '2023-04-19T19:11:37Z' }
  result: pass
  time-at-end:   { elapsed:     25.999, now: !!timestamp '2023-04-19T19:11:37Z' }
  test-runtime: 26.475
  threads:
  - thread: 0
    id: { logical:  0, package: 0, core:  0, thread: 0, family: 6, model: 0x55, stepping: 4, microcode: 0x2006e05, ppin: null }
    messages:
    - { level: warning, text: 'W> Test "intel_ifs_0" cannot be started at the moment, code: 0x600008200 image ID:  0x2 version: 0x172' }
```

File: tests/ifs/ifs.c [hunk 1]
```diff
@@ -207,15 +207,15 @@ static int scan_run_helper(struct test *test, int cpu)
...
                 ssize_t n = read_file(ifsfd, "details", result);
                 close(ifsfd);
 
                 if (n < 0) {
                         log_error("Test \"%s\" failed but could not retrieve error condition. Image ID: %s  version: %s", ifs_info->sys_dir, ifs_info->image_id, ifs_info->image_version);
                         return EXIT_FAILURE;
                 } else {
-                        if (sscanf(result, "%llx", &code) == 1 && is_result_code_skip(code)) {
+                        // Compare common driver error codes
+                        if (compare_error_codes(result, IFS_SW_TIMEOUT) || compare_error_codes(result, IFS_SW_PARTIAL_COMPLETION)) {
```
Comment (thiagomacieira) [inline tests/ifs/ifs.c:218]: You're parsing the string twice, now.

File: tests/ifs/sandstone_ifs.h [hunk 2]
```diff
@@ -22,12 +22,15 @@ typedef struct {
     char image_version[BUFLEN];
 } ifs_test_t;
 
-static bool is_result_code_skip(unsigned long long code)
+static bool compare_error_codes(char *c_code, unsigned long long expected)
```
Comment (thiagomacieira) [inline tests/ifs/sandstone_ifs.h:28]: Add `const` to `c_code`

File: tests/ifs/ifs.c [hunk 3]
```diff
@@ -226,8 +226,18 @@ static int scan_run_helper(struct test *test, int cpu)
...
+                {
                     strncpy(result, "unknown", BUFLEN);
-                log_warning("Test \"%s\" remains unstested, code: %s image ID: %s version: %s", ifs_info->sys_dir, result, ifs_info->image_id, ifs_info->image_version);
+                    log_warning("Test \"%s\" remains unstested, code: %s image ID: %s version: %s", ifs_info->sys_dir, result, ifs_info->image_id, ifs_info->image_version);
+                }
+                else
+                {
+                    if (compare_error_codes(result, IFS_SW_SCAN_CANNOT_START))
+                    {
+                        log_warning("Test \"%s\" cannot be started at the moment, code: %s image ID: %s version: %s", ifs_info->sys_dir, result, ifs_info->image_id, ifs_info->image_version);
```
Comment (thiagomacieira) [inline tests/ifs/ifs.c:237]: This can probably be downgraded to `log_info`

**Review by thiagomacieira — CHANGES_REQUESTED** [review]
Minor things, looking good.

**Comment by markdryan** [comment]
> Something that I'm still unable to do is marking test result as "skip" instead of "pass". Suggestions?
> 
> ```
> - test: ifs
>   time-at-start: { elapsed:      0.000, now: !!timestamp '2023-04-19T19:11:37Z' }
>   result: pass
>   time-at-end:   { elapsed:     25.999, now: !!timestamp '2023-04-19T19:11:37Z' }
>   test-runtime: 26.475
>   threads:
>   - thread: 0
>     id: { logical:  0, package: 0, core:  0, thread: 0, family: 6, model: 0x55, stepping: 4, microcode: 0x2006e05, ppin: null }
>     messages:
>     - { level: warning, text: 'W> Test "intel_ifs_0" cannot be started at the moment, code: 0x600008200 image ID:  0x2 version: 0x172' }
> ```

@AthenasJimenez not sure if this helps, but if every thread returns EXIT_SKIP from its run function the test will skip, even if EXIT_SUCCESS was returned from the init function.

**Review by thiagomacieira — APPROVED** [review]

**Pull request merged**